commands.js

```diff
@@ -1177,10 +1177,13 @@ exports.commands = {
 		if (cmd === 'roomauth1') userLookup = '\n\nTo look up auth for a user, use /userauth ' + target;
 		let targetRoom = room;
 		if (target) targetRoom = Rooms.search(target);
-		let unavailableRoom = targetRoom && (targetRoom !== room && (targetRoom.modjoin || targetRoom.staffRoom) && !user.can('makeroom'));
-		if (!targetRoom || unavailableRoom) return this.errorReply("The room '" + target + "' does not exist.");
+		if (!targetRoom) return this.errorReply("The room '" + target + "' does not exist.");
 		if (!targetRoom.auth) return this.sendReply("/roomauth - The room '" + (targetRoom.title ? targetRoom.title : target) + "' isn't designed for per-room moderation and therefore has no auth list." + userLookup);
 
+		let cannotJoin = targetRoom.isPrivate && targetRoom.modjoin && !targetRoom.auth[user.userid];
+		let unavailableRoom = (targetRoom !== room) && !user.inRooms.has(targetRoom.id) && (cannotJoin || targetRoom.staffRoom) && !user.can('makeroom');
```

Comment:
`(targetRoom !== room) && !user.inRooms.has(targetRoom.id)` simplifies to `!user.inRooms.has(targetRoom.id)`